### ISSUE TITLE
Photons in incorrect frame for CV models

### ISSUE THREAD

**Issue by saultyevil** [sirocco-rt/sirocco#790]
It seems for some of the example CV models that we are getting a bunch of these errors before any photon transport,

```
check_frame: move_phot :Photon (0) of type (0) not in frame 1
```

I think I've tracked this down to happening in functions like `phi()` and `hit_secondary()` , which are used (exclusively?) for binary systems. They both seems to be using photons which don't have a frame initialized before we use it, i.e. the global p_roche photon struct.

I think we probably want to say these photons are in observer frame, and I presume we also want to calculate the Roche potential (using `phi()`) in the observer frame (it being a boundary, which are defined in the observer frame).

**Comment by kslong** [comment]
I don't quite understand how this can happen, as described.  Please identify a .pf file where this is happening

**Comment by saultyevil** [comment]
I noticed this error it in the output of the Travis tests. It happens for cv_macro_benchmark, cv_standard and cv_standard_important. It doesn't happen to any of the other Travis models, so I think it's only happening for binary systems.

**Comment by saultyevil** [comment]
I think it's happening because we move photons around to find where L1 is or to calculate phi. For example, in phi we do this,

```c
stuff_phot (&p_roche, &pp);
move_phot (&pp, s);           /* So now we have the actuaal position of the photon relative to the WD */
```

But since p_roche doesn't have an initialized frame enumerator (F_OBSERVER or F_LOCAL), it's probably initialized with frame = 0, so the `check_frame` function throws the error when `pp` is in `move_phot()`.

I think the correct thing to do, would be to mark this photon as being in the observer frame since it's dealing with boundary stuff. I've made a PR to do this.

**Comment by saultyevil** [comment]
I believe this has been fixed with PR #791.
